creator-node/test/TranscodeDelegationManager.test.js

```diff
@@ -0,0 +1,274 @@
+const assert = require('assert')
+const sinon = require('sinon')
+const uuid = require('uuid')
+
+const TestUtils = require('./lib/utils')
+const { getLibsMock } = require('./lib/libsMock')
+
+const { logger: genericLogger } = require('../src/logging')
+const Utils = require('../src/utils')
+const TranscodeDelegationManager = require('../src/components/tracks/TranscodeDelegationManager')
+
+describe('test TranscodeDelegationManager', function () {
+  let libsMock, reqMock, logger
+  beforeEach(async () => {
+    libsMock = getLibsMock()
+    reqMock = {
+      logContext: { env: 'TranscodeDelegationManager test' },
+      fileDir: 'testFileDir',
+      fileName: 'testFileName.mp3',
+      fileNameNoExtension: 'testFileName',
+      uuid: uuid.v4(),
+      libs: libsMock
+    }
+    logger = genericLogger.child(reqMock.logContext)
+  })
+
+  afterEach(async function () {
+    sinon.restore()
+  })
+
+  // handOff()
+
+  it('If selecting random sps fails, return empty response', async function () {
+    sinon
+      .stub(TranscodeDelegationManager, 'selectRandomSPs')
+      .rejects(new Error('i have failed'))
+
+    const resp = await TranscodeDelegationManager.handOff(
+      { logContext: reqMock.logContext },
+      reqMock
+    )
+
+    assert.ok(TestUtils.deepEqual({}, resp))
+  })
+
+  // selectRandomSPs()
+
+  it('Selecting random SPs for transcode handoff works as expected', async function () {
+    const allSPs = libsMock.ethContracts.getServiceProviderList('content-node')
+    const allSPsSet = new Set(allSPs.map((sp) => sp.endpoint))
+
+    // Return 3 SPs for libs
+    let randomSPs = await TranscodeDelegationManager.selectRandomSPs(libsMock)
+    assert.strictEqual(randomSPs.length, 3)
+    randomSPs.forEach((sp) => {
+      assert.ok(allSPsSet.has(sp))
+    })
+
+    // Mock libs call to only return 1 sp
+    libsMock.ethContracts.getServiceProviderList = () => {
+      return allSPs.slice(0, 1)
+    }
+
+    // Return 1 SP for libs
+    let oneSP = await TranscodeDelegationManager.selectRandomSPs(libsMock)
+    assert.strictEqual(oneSP.length, 1)
+    assert.ok(allSPsSet.has(oneSP[0]))
+  })
+
+  // handOffToSP()
+
+  it('When handing off transcode request to an sp, if the health check fails, throw error', async function () {
+    sinon
+      .stub(TranscodeDelegationManager, 'fetchHealthCheck')
+      .rejects(new Error('failed to fetch health check response'))
+
+    let resp
+    try {
+      resp = await TranscodeDelegationManager.handOffToSp(logger, {
```

Comment:
I think there's too much overlap semantically between `handOff` and `handOffToSp` - can we make the delineation clearer?

Reply:
if i understand correctly `handOff` = `getTranscodedSegments` and `handOffToSp` = `enqueueTranscodeWithSp`  gah naming is hard but i don't think the first should be called `handoff` if we're going to have a subfunction of that that's called `handofftosp`...